File: java/src/main/java/it/giannibombelli/wsc2026/payment/application/commands/ExpirePayment.java
package it.giannibombelli.wsc2026.payment.application.commands;

import it.giannibombelli.wsc2026.common.utils.Require;

import it.giannibombelli.wsc2026.common.application.Command;
import it.giannibombelli.wsc2026.payment.domain.payment.PaymentId;

public record ExpirePayment(PaymentId aggregateId) implements Command<PaymentId> {
    public ExpirePayment {
        Require.requireArgument(aggregateId, "paymentId");
    }
}

File: typescript/src/payment/application/commands/expirePayment.ts
// Command per far scadere un pagamento.

import type { Command } from '@/common/application/command.js';
import type { PaymentId } from '../../domain/payment/paymentId.js';
import { requireArgument } from '@/common/utils/requireArgument.js';

export type ExpirePayment = Command<'ExpirePayment', PaymentId>;

export function expirePayment(aggregateId: PaymentId): ExpirePayment {
  requireArgument(aggregateId, 'paymentId');
  return { kind: 'ExpirePayment', aggregateId };
}
